image.yaml

```diff
@@ -1,232 +1,234 @@
+    example: "true"
+  - name: JAVA_MAX_MEM_RATIO
+    description: This is used to calculate a default maximal heap memory based on a containers restriction. If used in a container without any memory constraints for the container then this option has no effect. If there is a memory constraint then `-Xmx` is set to a ratio of the container available memory as set here. The default is `50` which means 50% of the available memory is used as an upper boundary. You can skip this mechanism by setting this value to `0` in which case no `-Xmx` option is added.
+    example: "50"
+  - name: JAVA_INITIAL_MEM_RATIO
+    description: This is used to calculate a default initial heap memory based the maximumal heap memory.  The default is `100` which means 100% of the maximal heap is used for the initial heap size.  You can skip this mechanism by setting this value to `0` in which case no `-Xms` option is added.
+    example: "100"
+  - name: JAVA_MAX_INITIAL_MEM
+    description: The maximum size of the initial heap memory, if the calculated default initial heap is larger then it will be capped at this value.  The default is 4096 MB.
+    example: "4096"
+  - name: JAVA_DIAGNOSTICS
+    description: Set this to get some diagnostics information to standard output when things are happening. **Disabled by default.**
+    example: "true"
+  - name: GC_MIN_HEAP_FREE_RATIO
+    description: Minimum percentage of heap free after GC to avoid expansion.
+    example: "20"
+  - name: GC_MAX_HEAP_FREE_RATIO
+    description: Maximum percentage of heap free after GC to avoid shrinking.
+    example: "40"
+  - name: GC_TIME_RATIO
+    description: Specifies the ratio of the time spent outside the garbage collection (for example, the time spent for application execution) to the time spent in the garbage collection.
+    example: "4"
+  - name: GC_ADAPTIVE_SIZE_POLICY_WEIGHT
+    description: The weighting given to the current GC time versus previous GC times.
+    example: "90"
+  - name: GC_MAX_METASPACE_SIZE
+    description: The maximum metaspace size.
+    example: "100"
+  - name: "CONTAINER_HEAP_PERCENT"
+    example: 0.5
+    description: Deprecated.  See JAVA_MAX_MEM_RATIO.
+  - name: "INITIAL_HEAP_PERCENT"
+    example: 0.5
+    description: Deprecated.  See JAVA_INITIAL_MEM_RATIO.
+  - name: "PROBE_DISABLE_BOOT_ERRORS_CHECK"
+    example: "true"
+    description: Disable the boot errors check in the probes.
+
 ports:
-    - value: 8080
-    - value: 8443
-    - value: 8787
-      expose: false
-    - value: 11211
-    - value: 11222
-    - value: 11333
+  - value: 8080
+  - value: 8443
+  - value: 8787
+    expose: false
+  - value: 11211
+  - value: 11222
+  - value: 11333
+
 modules:
-      repositories:
-          - name: cct_module
-            git:
-                  url: https://github.com/jboss-openshift/cct_module.git
-                  ref: master
-          - name: datagrid.72.modules
-            path: modules/datagrid/72
```

Comment:
As we no longer use any of the modules in `modules/datagrid/72`, we should delete them. Also, I think we should remove the version dir completely. We're using branches to differentiate between versions, therefore by having `72` etc, it means that every time we move to the next DG version we have to mess up our git diff as files are moved. If we remove the version, this is no longer an issue.

Also, a general comment about the module structure. I like the idea of having the `common` directory to show what has come from `cct_modules`, however I don't think this should be a subdir of datagrid. Instead how about the following structure:

```
modules/
   - datagrid/
      - configuration
      - distribution
      - launch
   - common/
      - datagrid-7-openshift
     ....
```

Long term as we simplify the modules inherited from `cct_modules` and remove the scripts/configuration options that are redundant to DG we can reduce the size of the common dir.